eth/stagedsync/stage_polygon_sync.go

```diff
@@ -378,15 +378,15 @@ func (s polygonSyncStageCheckpointStore) GetEntity(ctx context.Context, id uint6
 		return nil
 	})
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 	if r.err != nil {
-		return nil, r.err
+		return nil, false, r.err
 	}
 
 	var c heimdall.Checkpoint
 	err = json.Unmarshal(r.v, &c)
-	return &c, err
+	return &c, true, err
```

Comment:
Returning `true` here is not correct. (Hint: You would need to check how `checkpointReader.Checkpoint` works and handle whatever it returns for the "not found" case and map it to the current new expectation of `GetEntity`.)

Reply:
got confused - ignore me :)